### PR TITLE
Refactor edit renderers [SCI-4064]

### PR DESCRIPTION
**Pull request by urbanrotnik**
Jira ticket: [SCI-4064](https://biosistemika.atlassian.net/browse/SCI-4064)
Also
Jira ticket: [SCI-4063](https://biosistemika.atlassian.net/browse/SCI-4063)
Jira ticket: [SCI-4062](https://biosistemika.atlassian.net/browse/SCI-4062)
Jira ticket: [SCI-4061](https://biosistemika.atlassian.net/browse/SCI-4061)

### What was done
- View renderers for DateTime column types
- Edit renderer for List ColumnType
- Edit renderer for Statuses
- Edit renderer for Assets
- New renderer for Assets, Status, Lists

### CODE REVIEW

File: app/assets/javascripts/application.js.erb [hunk 1]
```diff
@@ -44,6 +44,7 @@
 //= require protocols/header
 //= require marvinjslauncher
 //= require_directory ./repositories/renderers
+//= require_directory ./repositories/renderers/columns
```
Comment (okriuchykhin) [inline app/assets/javascripts/application.js.erb]: Is it needed here?

File: app/models/repository_cell.rb [hunk 2]
```diff
@@ -80,8 +80,20 @@ def self.create_with_value!(row, column, data, user)
   private
 
   def repository_column_data_type
-    if !repository_column || value_type != repository_column.data_type
+    if !repository_column || value_type != repository_column.data_type && !date_time_compatibility_checks
       errors.add(:value_type, 'must match column data type')
     end
   end
+
+  def date_time_compatibility_checks
```
Comment (okriuchykhin) [inline app/models/repository_cell.rb]: Why do you need it?

File: app/serializers/repository_datatable/repository_date_time_range_value_serializer.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module RepositoryDatatable
+  class RepositoryDateTimeRangeValueSerializer < ActiveModel::Serializer
+    attributes :value, :value_type
+
+    def value
+      date_time_format = @instance_options[:user].settings[:date_format] + ', %H:%M'
```
Comment (okriuchykhin) [inline app/serializers/repository_datatable/repository_date_time_range_value_serializer.rb]: Use I18n.l here

File: app/views/repository_rows/index.json.jbuilder [hunk 4]
```diff
@@ -6,5 +6,6 @@ json.data do
                                   @repository,
                                   @columns_mappings,
                                   @repository.team,
-                                  @assigned_rows)
+                                  @assigned_rows,
+                                  @current_user)
```
Comment (okriuchykhin) [inline app/views/repository_rows/index.json.jbuilder]: Remove user parameter

File: app/serializers/repository_datatable/repository_date_time_value_serializer.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module RepositoryDatatable
+  class RepositoryDateTimeValueSerializer < ActiveModel::Serializer
+    attributes :value, :value_type
+
+    def value
+      date_time_format = @instance_options[:user].settings[:date_format] + ', %H:%M'
```
Comment (okriuchykhin) [inline app/serializers/repository_datatable/repository_date_time_value_serializer.rb]: Use I18n.l here

File: app/serializers/repository_datatable/repository_date_value_serializer.rb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module RepositoryDatatable
+  class RepositoryDateValueSerializer < ActiveModel::Serializer
+    attributes :value, :value_type
+
+    def value
+      date_time_format = @instance_options[:user].settings[:date_format]
```
Comment (okriuchykhin) [inline app/serializers/repository_datatable/repository_date_value_serializer.rb]: Use I18n.l here

File: app/serializers/repository_datatable/repository_time_range_value_serializer.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module RepositoryDatatable
+  class RepositoryTimeRangeValueSerializer < ActiveModel::Serializer
+    attributes :value, :value_type
+
+    def value
+      date_time_format = '%H:%M'
```
Comment (okriuchykhin) [inline app/serializers/repository_datatable/repository_time_range_value_serializer.rb]: Use I18n.l here

File: app/serializers/repository_datatable/repository_time_value_serializer.rb [hunk 8]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module RepositoryDatatable
+  class RepositoryTimeValueSerializer < ActiveModel::Serializer
+    attributes :value, :value_type
+
+    def value
+      date_time_format = '%H:%M'
```
Comment (okriuchykhin) [inline app/serializers/repository_datatable/repository_time_value_serializer.rb]: Use I18n.l here

File: app/serializers/repository_datatable/repository_date_range_value_serializer.rb [hunk 9]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module RepositoryDatatable
+  class RepositoryDateRangeValueSerializer < ActiveModel::Serializer
+    attributes :value, :value_type
+
+    def value
+      date_time_format = @instance_options[:user].settings[:date_format]
```
Comment (okriuchykhin) [inline app/serializers/repository_datatable/repository_date_range_value_serializer.rb]: Use I18n.l here

File: app/helpers/repository_datatable_helper.rb [hunk 10]
```diff
@@ -66,7 +67,8 @@ def default_table_columns
     Constants::REPOSITORY_TABLE_DEFAULT_STATE['columns'].to_json
   end
 
-  def display_cell_value(cell)
-    "RepositoryDatatable::#{cell.value_type}Serializer".constantize.new(cell).serializable_hash
+  def display_cell_value(cell, user)
```
Comment (okriuchykhin) [inline app/helpers/repository_datatable_helper.rb]: remove user

File: app/helpers/repository_datatable_helper.rb [hunk 11]
```diff
@@ -36,7 +37,7 @@ def prepare_row_columns(repository_rows,
       # Add custom columns
       record.repository_cells.each do |cell|
         row[columns_mappings[cell.repository_column.id]] =
-          display_cell_value(cell)
+          display_cell_value(cell, current_user)
```
Comment (okriuchykhin) [inline app/helpers/repository_datatable_helper.rb]: remove user

File: app/helpers/repository_datatable_helper.rb [hunk 12]
```diff
@@ -7,7 +7,8 @@ def prepare_row_columns(repository_rows,
                           repository,
                           columns_mappings,
                           _team,
-                          assigned_rows)
+                          assigned_rows,
+                          current_user)
```
Comment (okriuchykhin) [inline app/helpers/repository_datatable_helper.rb]: remove user

File: app/controllers/repository_status_items_controller.rb [hunk 13]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class RepositoryStatusItemsController < ApplicationController
+  before_action :load_vars, only: :search
```
Comment (okriuchykhin) [inline app/controllers/repository_status_items_controller.rb]: remove :only as it also does basic permission checks

File: app/assets/javascripts/application.js.erb (hunk 1 above)
Comment (urbanrotnik) [reply app/assets/javascripts/application.js.erb]: Yes or we can use `require_tree` for `.repositories/renderers`

**Pull request merged**